cmd/experimental/kjobctl/test/e2e/slurm_test.go

```diff
@@ -0,0 +1,315 @@
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		defer script.Close()
+		defer os.Remove(script.Name())
+
+		ginkgo.By("Prepare script", func() {
+			_, err := script.WriteString("#!/bin/bash\nwhile true; do printenv | grep SLURM_ > /env.out; sleep 0.25; done")
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		})
+
+		var output []byte
+
+		ginkgo.By("Create slurm", func() {
+			cmdArgs := []string{"create", "slurm", "-n", ns.Name, "--profile", profile.Name}
+			cmdArgs = append(cmdArgs, args...)
+			cmdArgs = append(cmdArgs, "--", script.Name())
+			cmdArgs = append(cmdArgs, slurmArgs...)
+			cmd := exec.Command(kjobctlPath, cmdArgs...)
+			output, err = util.Run(cmd)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred(), "%s: %s", err, output)
+			gomega.Expect(output).NotTo(gomega.BeEmpty())
+		})
+
+		jobName, configMapName, serviceName := parseSlurmCreateOutput(output, profile.Name)
+		gomega.Expect(jobName).NotTo(gomega.BeEmpty())
+		gomega.Expect(configMapName).NotTo(gomega.BeEmpty())
+		gomega.Expect(serviceName).NotTo(gomega.BeEmpty())
+
+		job := &batchv1.Job{}
+		configMap := &corev1.Service{}
+		service := &corev1.Service{}
+
+		ginkgo.By("Check slurm is created", func() {
+			gomega.Expect(k8sClient.Get(ctx, client.ObjectKey{Namespace: ns.Name, Name: jobName}, job)).To(gomega.Succeed())
+			gomega.Expect(ptr.Deref(job.Spec.Completions, 1)).To(gomega.Equal(expectCompletions))
+			gomega.Expect(ptr.Deref(job.Spec.Parallelism, 1)).To(gomega.Equal(expectParallelism))
+
+			gomega.Expect(k8sClient.Get(ctx, client.ObjectKey{Namespace: ns.Name, Name: configMapName}, configMap)).To(gomega.Succeed())
+			gomega.Expect(k8sClient.Get(ctx, client.ObjectKey{Namespace: ns.Name, Name: serviceName}, service)).To(gomega.Succeed())
+		})
+
+		var firstPod *corev1.Pod
+
+		for completionIndex, expectPod := range expectPods {
+			podList := &corev1.PodList{}
+
+			ginkgo.By(fmt.Sprintf("Wait for pod with completion index %d is running", completionIndex), func() {
+				gomega.Eventually(func(g gomega.Gomega) {
+					g.Expect(k8sClient.List(ctx, podList, client.InNamespace(ns.Name), client.MatchingLabels(map[string]string{
+						BatchJobNameLabel:            job.Name,
+						BatchJobCompletionIndexLabel: strconv.Itoa(completionIndex),
+					}))).Should(gomega.Succeed())
+					g.Expect(podList.Items).Should(gomega.HaveLen(1))
+					g.Expect(podList.Items[0].Status.Phase).To(gomega.Equal(corev1.PodRunning))
+					if completionIndex == 0 {
+						firstPod = &podList.Items[0]
+					}
+				}, util.LongTimeout, util.Interval).Should(gomega.Succeed())
+			})
+
+			pod := podList.Items[0]
+
+			for containerName, expectContainerVars := range expectPod {
+				ginkgo.By(fmt.Sprintf("Check env variables in index %d and container name %s", completionIndex, containerName), func() {
+					gomega.Eventually(func(g gomega.Gomega) {
+						cmd := exec.Command("kubectl", "exec", "-c", containerName, "-n", ns.Name, pod.Name, "--", "cat", "/env.out")
```

Comment:
can't we do this with a go client? there is no guarantee the default kubectl profile points to the cluster in which the e2e tests are running.

(I think the interactive mode is doing it)

Reply:
> (I think the interactive mode is doing it)

We are not using it on interactive. Also we don't have e2e test for interactive.

Reply:
Done